### PR TITLE
Add Feast CLI / python SDK documentation

### PR DESCRIPTION
**Pull request by romanwozniak**
In this PR: 
- [x] Removed all mentions of `granularity` from the documentation / proto (because it was removed from the codebase in https://github.com/gojek/feast/issues/17)
- [x] Picked up some WIP Feast CLI docs from @thirteen37 and updated them with information about import job spec registration
- [x] Added a quick start guide for feast python SDK (how to install it, register entity/features, ingest features into feast, retrieve training/serving data)

### CODE REVIEW

File: docs/specs.md [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+|`name`|string|lower snake case, e.g: `age_prediction`| short name of a feature |
+|`owner`|string|use email or any unique identifier| `owner` is mainly be used to inform user who is responsible of maintaining the feature |
+|`description`|string| Keep the description concise, more information about the feature can be provided as documentation linked by `uri` | human readable description of a feature |
+|`valueType`| Enum(one of: `BYTES`,`STRING`,`INT32`,`INT64`,`DOUBLE`,`FLOAT`, `BOOL`, `TIMESTAMP`)| N.A.| Value type of the feature |
+|`group`| string | lower snake case | feature group inherited by this feature.|
+| `tags` | List of string | N.A. | Free form grouping |
+| `options` | Key value string | N.A. | Option is used for extendability of feature spec|
+| `dataStores`| See below table | See below table | Storage used to store the feature|
+
+#### DataStores
```
Comment (zhilingc) [inline docs/specs.md]: declaration of datastores has been deprecated as of pr #159

File: examples/values.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    maxNumberOfThread: 128
+    redisPool: 
+      maxIdle: 16
+      maxSize: 128
+    timeout: 5
+  image: 
+    pullPolicy: IfNotPresent
+    registry: gcr.io/kf-feast
+    repository: feast-serving
+    tag: "1234"
```
Comment (zhilingc) [inline examples/values.yaml]: you can set this tag to our latest release (0.1.0)

File: docs/specs.md (hunk 1 above)
Comment (romanwozniak) [reply docs/specs.md]: Oh great, I didn't know that. Does it make sense to remove this from python SDK's `Importer` API as well?

File: examples/values.yaml (hunk 2 above)
Comment (romanwozniak) [reply examples/values.yaml]: Done, thanks @zhilingc

File: docs/specs.md (hunk 1 above)
Comment (davidheryanto) [reply docs/specs.md]: Yes I think we should

**Comment by zhilingc** [comment]
/lgtm

Thanks Roman!

**Pull request merged**